ambry-frontend/src/main/java/com/github/ambry/frontend/FrontendRestRequestService.java

```diff
@@ -290,6 +294,9 @@ public void handleGet(final RestRequest restRequest, final RestResponseChannel r
           && NamedBlobPath.parse(requestPath, restRequest.getArgs()).getBlobName() == null) {
         namedBlobListHandler.handle(restRequest, restResponseChannel,
             (result, exception) -> submitResponse(restRequest, restResponseChannel, result, exception));
+      } else if (RestUtils.getBooleanHeader(restRequest.getArgs(), ENABLE_DATASET_VERSION_LISTING, false)) {
+        listDatasetVersionHandler.handle(restRequest, restResponseChannel,
+            (result, exception) -> submitResponse(restRequest, restResponseChannel, result, exception));
```

Comment:
This predicate is too simple, can we add some condition on the url as well.

Reply:
Added one more condition to check if the version info is empty. But the major condition is ENABLE_DATASET_VERSION_LISTING because it's hard for us to detect if they provide the version or not, since the datasetName could include "/".